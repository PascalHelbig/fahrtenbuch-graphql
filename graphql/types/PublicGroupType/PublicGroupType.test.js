const { makeExecutableSchema, addMockFunctionsToSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const PublicGroupType = require('./PublicGroupType');

it('has to be a valid schema', () => {
  const SchemaDefinition = 'schema { query: PublicGroup }';
  const schema = makeExecutableSchema({ typeDefs: [SchemaDefinition, PublicGroupType] });
  addMockFunctionsToSchema({
    schema,
    mocks: {
      PublicGroup: () => ({
        id: () => '1337',
        name: () => 'Club Name',
        is_club: () => true,
      }),
    },
  });

  const query = `
    query {
      id
      name
      is_club
    }
  `;

  return graphql(schema, query).then(result =>
    expect(result).toMatchSnapshot()
  );
});
