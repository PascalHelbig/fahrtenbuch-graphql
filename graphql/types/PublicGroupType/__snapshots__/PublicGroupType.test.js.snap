exports[`test has to be a valid schema 1`] = `
Object {
  "data": Object {
    "id": "1337",
    "is_club": true,
    "name": "Club Name",
  },
}
`;
